### PR TITLE
v4: drop Node.js < 10 and dev dependency updates

### PR DESCRIPTION
**Pull request by XhmikosR**
This drops the engines version from package.json and updates all devDependencies. Also I updated Jekyll to v4.0.0 which is only slightly faster but better than nothing.

In the past we considered such changes breaking ones. I don't think they should be considered breaking ones since they are development only changes. And from now on we should be more lax on master branch anyway.

/CC @twbs/team for feedback.

PS. I will probably rebase this later with further dep updates.

Refs #29821 

TODO:

- [x] Adapt v4-dev branch protection rules
- [x] Switch to terser
- [x] look if we can update jekyll-toc
- [x] Make sure docs are fine: <https://deploy-preview-30006--twbs-bootstrap.netlify.com/>

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -52,6 +52,9 @@ jobs:
       - name: Set up Bundler
         run: gem install bundler -v "~> 1.17"
 
+      - name: Update rubygems
```
Comment (XhmikosR) [inline .github/workflows/test.yml]: Note to self: check if we need this

**Comment by MartijnCuppens** [comment]
>  I don't think they should be considered breaking ones since they are development only changes

I agree

**Comment by ysds** [comment]
I have not checked the diffs, but I agree with the node v6 drop.

**Comment by XhmikosR** [comment]
Alright, I guess only a few todo are left and @mdo's review

File: .github/workflows/test.yml (hunk 1 above)
Comment (XhmikosR) [reply .github/workflows/test.yml]: This is still needed with Ruby 2.4.

**Comment by XhmikosR** [comment]
This is ready from my side. Current TODO (also added in the OP)

- [ ] Wait for a new stylelint-config-twbs-bootstrap version for stylelint 13.0.0 compatibility
- [ ] Adapt v4-dev branch protection rules before the merge

**Comment by XhmikosR** [comment]
Must be https://github.com/browserslist/browserslist/issues/439

**Comment by XhmikosR** [comment]
Alright, I think this is ready.

/CC @mdo please check it out and let me know how to proceed.

On a side note, this results in smaller JS dist files:
[before](https://bundlesize-store.now.sh/build?info=%7B%22files%22%3A%5B%7B%22maxSize%22%3A7168%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-grid.css%22%2C%22size%22%3A7083%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A6400%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-grid.min.css%22%2C%22size%22%3A6210%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A2048%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-reboot.css%22%2C%22size%22%3A1706%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A2048%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-reboot.min.css%22%2C%22size%22%3A1597%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A25856%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap.css%22%2C%22size%22%3A25688%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A23808%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap.min.css%22%2C%22size%22%3A23618%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A48640%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.bundle.js%22%2C%22size%22%3A48327%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A22784%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.bundle.min.js%22%2C%22size%22%3A22744%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A25600%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.js%22%2C%22size%22%3A25326%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A16128%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.min.js%22%2C%22size%22%3A15886%2C%22compression%22%3A%22gzip%22%7D%5D%2C%22repo%22%3A%22twbs%2Fbootstrap%22%2C%22branch%22%3A%22v4-dev%22%2C%22commit_message%22%3A%22%22%2C%22sha%22%3A%2269f36286f97018adc59e5d2ed64d7ea28dd9b5dc%22%7D) - [after](https://bundlesize-store.now.sh/build?info=%7B%22files%22%3A%5B%7B%22maxSize%22%3A7168%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-grid.css%22%2C%22size%22%3A7083%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A6400%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-grid.min.css%22%2C%22size%22%3A6210%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A2048%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-reboot.css%22%2C%22size%22%3A1706%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A2048%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap-reboot.min.css%22%2C%22size%22%3A1597%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A25856%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap.css%22%2C%22size%22%3A25688%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A23808%2C%22path%22%3A%22.%2Fdist%2Fcss%2Fbootstrap.min.css%22%2C%22size%22%3A23618%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A48640%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.bundle.js%22%2C%22size%22%3A48327%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A22784%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.bundle.min.js%22%2C%22size%22%3A22608%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A25600%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.js%22%2C%22size%22%3A25326%2C%22compression%22%3A%22gzip%22%7D%2C%7B%22maxSize%22%3A16128%2C%22path%22%3A%22.%2Fdist%2Fjs%2Fbootstrap.min.js%22%2C%22size%22%3A15756%2C%22compression%22%3A%22gzip%22%7D%5D%2C%22repo%22%3A%22twbs%2Fbootstrap%22%2C%22branch%22%3A%22v4-dev-xmr-drop-node-6%22%2C%22commit_message%22%3A%22%22%2C%22sha%22%3A%227ea8466436c2ce6563bffba631599cc69d68ff83%22%7D)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        node: [6, 8, 12]
+        node: [10, 12]
```
Comment (MartijnCuppens) [inline .github/workflows/test.yml]: Did you drop **node 8** here too?

Comment (XhmikosR) [reply .github/workflows/test.yml]: Of course. It's also EOL. I used what we use in master which is the current LTS

Comment (MartijnCuppens) [reply .github/workflows/test.yml]: Ok, I adjusted the PR title

**Review by mdo — APPROVED** [review]

**Comment by XhmikosR** [comment]
@mdo can you adapt the v4-dev branch protection rules? I can't merge without this.

**Comment by mdo** [comment]
@XhmikosR Done! Sorry for the delay, node 6 and 8 no longer required.

**Comment by XhmikosR** [comment]
Cool, I'll have another look and merge it after that. So happy to untie us from unsupported Node.js versions :)

**Review by Johann-S — APPROVED** [review]

**Pull request merged**